### ISSUE TITLE
feature: Backlinks

### ISSUE THREAD

**Issue by pawamoy** [mkdocstrings/python#153]
### Is your feature request related to a problem? Please describe.
I'm having fun looking at [matrix-nio's API](https://matrix-nio.readthedocs.io/en/latest/nio.html#nio.responses.JoinedRoomsResponse), and also having trouble finding how to get the details of a room.

I resorted to searching for "room name" into the big HTML page, and ended up on this class: https://matrix-nio.readthedocs.io/en/latest/nio.html#nio.rooms.MatrixRoom, which seems to provide the info I want.

However I don't how I'm supposed to get back instances of this class (`client.joined_rooms` only returns a list of room ids as strings).

So I thought, "would be nice if there were links in this class docs showing me what functions/methods return instances of it". Backlinks!

### Describe the solution you'd like
A way to display backlinks in the rendered docs:

- For a class: which functions/methods return instances of it.
- For anything: which functions/methods uses it. Not possible yet as we don't scan function bodies.
- ?

### Describe alternatives you've considered
Manual backlinks?

### Additional context
This will probably need changes in Griffe.

**Comment by mdrxy** [comment]
Are there any known issues with backlinks? Have it enabled (and insiders installed) and am not seeing any across our docs at https://reference.langchain.com/python/

**Comment by pawamoy** [comment]
Backlinks are only rendered under API symbols injected with mkdocstrings, not within regular Markdown sections. Is that what you expected?

**Comment by mdrxy** [comment]
@pawamoy yes, that's what I expect (if I am understanding you correctly). Our docs are built by injecting w/ mkdocstrings and `backlinks` are [enabled](https://github.com/langchain-ai/docs/blob/main/reference/python/mkdocs.yml#L220). See [this source file](https://github.com/langchain-ai/docs/blob/main/reference/python/docs/langchain_core/language_models.md?plain=1) as an example (and it's corresponding [built page](https://reference.langchain.com/python/langchain_core/language_models/)). I'd expect to see many backlinks [on `BaseChatModel`](https://reference.langchain.com/python/langchain_core/language_models/#langchain_core.language_models.chat_models.BaseChatModel) but do not (such as [`BaseChatOpenAI`](https://reference.langchain.com/python/integrations/langchain_openai/BaseChatOpenAI/#langchain_openai.chat_models.base)).

Could you provide a toy example showing a scenario in which a backlink should be created? I think the lack of an example in the docs is contributing to the confusion here.

**Comment by pawamoy** [comment]
Could you open a new issue for this please? And if possible provide a minimal reproduction example? Your docs repo is quite big, even just the Python part 😅
